pipeline/stage02_processing/scripts/spatial_downsampling.py

```diff
@@ -1,30 +1,24 @@
 """
-Substract the background of a given dataset by substracting the mean of each channel.
+Spatial downsampling of the input dataset
 """
+
 import numpy as np
 import matplotlib.pyplot as plt
 import argparse
 import os
+import neo
 import quantities as pq
 from skimage import data, io, filters, measure
 from utils import determine_spatial_scale, load_neo, write_neo, save_plot, \
-                  none_or_str
+                  none_or_str, AnalogSignal2ImageSequence, ImageSequence2AnalogSignal
 
 
-def spatial_smoothing(asig, MACRO_PIXEL_DIM, DIM_X, DIM_Y):
+def spatial_smoothing(images, MACRO_PIXEL_DIM):
 
-    print("         Spacial smoothing...")
     # Now we need to reduce the noise from the images by performing a spatial smoothing
-    img_collection_reduced = []
-    for elem in range(0, len(asig)): #for each image
-        img = np.reshape(asig[elem], (int(DIM_X),int(DIM_Y)))
-        img_reduced = measure.block_reduce(img, (MACRO_PIXEL_DIM, MACRO_PIXEL_DIM), np.mean)
-        img_reduced = np.reshape(img_reduced,(1, len(img_reduced)*len(img_reduced)))
-        img_collection_reduced.append(img_reduced[0,:])
-    print("         Images reduced")
+    images = measure.block_reduce(images, (1, MACRO_PIXEL_DIM, MACRO_PIXEL_DIM), np.mean, cval = np.median(images))
 
-    #----------------------
-    return img_collection_reduced
+    return images
```

Comment:
The input of the function is an imagesequence object. So, it would be more consistent to also move the creation of the spatially downsampled imagesequence object into the function and also return an imagesequence object.